### PR TITLE
Add toggles for multiple builtin wish lists, allow incrementally adding/removing URLs

### LINKED ISSUE #9421: It isn't entirely clear when a pre-made wishlist has been selected
### Proposed change

If you select "choosy_voltron" or "voltron (default)" from the pre-made list it is somewhat unclear on subsequent visits whether the displayed button is showing the selected wishlist, or the non-selected one.
image

I would like to update the pre-made list selection to this, when one is selected: 
image

### How does this fit into your workflow?

This will just make it clearer when one of the pre-made lists is selected at a glance.

### PR DESCRIPTION
**Pull request by robojumper**
Fixes #9421, supercedes #9422

grafik

Tries to do this without too many changes to the wish lists internals - there are some annoyances like the fact that loading automatically happens when toggling something and it blocks the main thread...

### CODE REVIEW

File: src/app/settings/WishListSettings.tsx [hunk 1]
```diff
@@ -62,19 +46,15 @@ export default function WishListSettings() {
...
     const reader = new FileReader();
     reader.onload = async () => {
       if (reader.result && typeof reader.result === 'string') {
-        const wishListAndInfo = toWishList(reader.result);
+        const wishListAndInfo = toWishList([undefined, reader.result]);
+        if (wishListAndInfo.wishListRolls.length) {
+          dispatch(clearWishLists());
+        }
+        // Still attempt to store even with 0 rolls to show an error message
         dispatch(transformAndStoreWishList(wishListAndInfo));
```
Comment (bhollis) [inline src/app/settings/WishListSettings.tsx:58]: Should `transformAndStoreWishList` do the clearing of wishlists?

File: src/app/wishlists/wishlist-file.ts [hunk 2]
```diff
@@ -22,7 +22,9 @@ const notesLabel = '//notes:';
  * one or more wish list text files, deduplicating within
  * and between lists.
  */
-export function toWishList(...fileTexts: string[]): WishListAndInfo {
+export function toWishList(
+  ...files: [url: string | undefined, contents: string][]
```
Comment (bhollis) [inline src/app/wishlists/wishlist-file.ts:26]: Would it be better to take a list of objects than a list of tuples?

**Review by bhollis — APPROVED** [review]
The thing where it blocks the main thread when you try to check the box is pretty bad - I ended up clicking multiple times and turning the voltron list on and off a few times.

A couple other things that this seems to pave the way for that might be nice, but don't have to be part of this PR:
1. It'd be great if we could break out all the individual voltron sources into this list so you could individually turn them on and off. By default I guess they'd all be checked? This starts to edge into my (very paused) wishlist revamp where wishlists can compose other wishlists.
2. It'd be neat if this interface allowed for having both wishlists from URLs and wishlists from files.

**Comment by robojumper** [comment]
> It'd be neat if this interface allowed for having both wishlists from URLs and wishlists from files.

The main complication is that the "transformed" wish list is always a deduplicated list of rolls as a result of parsing wish list sources in order if there is a settings URL (where DIM takes care of fetching and updating that from time to time) or a bunch of rolls from the file import (no URL, DIM doesn't touch it at all). Combining the two would require storing file rolls separately too to re-combine when any list updates. Also we currently can't sync wish list files via DIM Sync...

**Comment by bhollis** [comment]
Got it, sounds like it's not worth shooting for until the forever-delayed wishlist revamp.

**Comment by bhollis** [comment]
Happy for this to go in as-is

**Pull request merged**